scripts/snowflake_generate_grant_all/main.go

```diff
@@ -0,0 +1,202 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"sort"
+	"strings"
+
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/provider"
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/resources"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/errgo.v2/fmt/errors"
+)
+
+const (
+	topLvlComment string = `
+	NOTE: Generated by scripts/snowflake_generate_grant_all.
+	      Changes made directly to this file will be overwritten.
+				Make improvements there so everyone can benefit.`
+
+	perPrivTypeVarName       string = "per_privilege_grants"
+	snowflakeProviderVersion string = "> 0.18.0"
+)
+
+type Variable struct {
+	TType       string      `json:"type"`
+	Description string      `json:"description"`
+	Default     interface{} `json:"default"`
+}
+
+type ModuleTemplate struct {
+	Comment   string                 `json:"//,omitempty"`
+	Variables map[string]Variable    `json:"variable,omitempty"`
+	Locals    map[string]interface{} `json:"locals,omitempty"`
+
+	// resource type: resource name: arguments
+	Resources map[string]map[string]map[string]interface{} `json:"resource,omitempty"`
+
+	// required_providers: provider_name: version
+	Terraform map[string]map[string]string `json:"terraform,omitempty"`
+}
+
+func main() {
+	err := exec()
+	if err != nil {
+		logrus.Fatal(err)
+		return
+	}
+}
+
+func exec() error {
+	grants := provider.GetGrantResources()
+	for name, grant := range grants {
+		tf, err := generateModule(name, grant)
+		if err != nil {
+			return err
+		}
+		err = writeModule(name, tf)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Assume we're running from this directory
+func writeModule(name string, tf []byte) error {
+	moduleName := fmt.Sprintf("%s-all", strings.ReplaceAll(name, "_", "-"))
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	// make sure dir is there
+	moduleDir := path.Join(cwd, "..", "..", moduleName)
+	err = os.MkdirAll(moduleDir, 0755)
+	if err != nil {
+		return err
+	}
+
+	// write the file
+	return ioutil.WriteFile(path.Join(moduleDir, "main.tf.json"), tf, 0644)
+}
+
+func generateModule(name string, grant *resources.TerraformGrantResource) ([]byte, error) {
+	privileges := grant.ValidPrivs.ToList()
+	sort.Strings(privileges)
+
+	m := &ModuleTemplate{
+		Comment:   topLvlComment,
+		Variables: map[string]Variable{},
+		Locals: map[string]interface{}{
+			"privileges": privileges,
+		},
+		Terraform: map[string]map[string]string{
+			"required_providers": {
+				"snowflake": snowflakeProviderVersion,
```

Comment:
This probably needs to be updated each time we generate, since code with new privileges could be generated.

Reply:
good catch, thanks

Reply:
this might require some changes in the provider side to do properly so I might defer it to a follow-up change. This PR is already getting big for my liking